tpm/tpm_test.go

```diff
@@ -20,57 +20,52 @@ import (
 func TestIsDevNode(t *testing.T) {
 	for _, test := range []struct {
 		path   string
-		kind   string
+		kind   tpm.DevNodeKind
 		expect bool
 	}{
 		{
 			path:   "/dev/tpm0",
-			kind:   "tpm",
+			kind:   tpm.DevNodeUnmanaged,
 			expect: true,
 		},
 		{
 			path:   "/dev/tpm1",
-			kind:   "tpm",
+			kind:   tpm.DevNodeUnmanaged,
 			expect: true,
 		},
 		{
 			path:   "/dev/tpmrm0",
-			kind:   "tpmrm",
+			kind:   tpm.DevNodeManaged,
 			expect: true,
 		},
 		{
 			path:   "/dev/tpmrm1",
-			kind:   "tpmrm",
+			kind:   tpm.DevNodeManaged,
 			expect: true,
 		},
 		{
 			path:   "/dev/tpm0",
-			kind:   "tpmrm",
+			kind:   tpm.DevNodeManaged,
 			expect: false,
 		},
 		{
 			path:   "/dev/tpmrm0",
-			kind:   "tpm",
-			expect: false,
-		},
-		{
-			path:   "/dev/tpmrm0",
-			kind:   "tpmrm0",
+			kind:   tpm.DevNodeUnmanaged,
 			expect: false,
 		},
 		{
 			path:   "tpmrm0",
-			kind:   "tpmrm",
+			kind:   tpm.DevNodeManaged,
 			expect: false,
 		},
 	} {
-		t.Run("whether "+test.path+" is a "+test.kind, func(t *testing.T) {
+		t.Run("whether "+test.path+" is a "+test.kind.PathPrefix(), func(t *testing.T) {
 			if got, expect := tpm.IsDevNode(test.path, test.kind), test.expect; got != expect {
```

Comment:
These tests will fail if the device doesn't actually exist. How can we fake the result of `os.Stat`?